bee-ternary/src/convert.rs

```diff
@@ -0,0 +1,369 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use crate::{Btrit, RawEncoding, RawEncodingBuf, TritBuf, Trits, Utrit};
+use num_traits::{AsPrimitive, CheckedAdd, FromPrimitive, Signed, Num};
+use std::convert::TryFrom;
+
+/// An error that may be produced during numeric conversion
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// The trit slice didn't contain enough trits to be considered a numeric value.
+    Empty,
+    /// The trit slice has a numeric value that's too big to contain within the requested numeric
+    /// type.
+    TooBig,
+}
+
+// TODO: Find a way to implement this without conflicting impls
+// impl<I, T: RawEncodingBuf> From<I> for TritBuf<T>
+// where
+//     T::Slice: RawEncoding<Trit = Btrit>,
+//     I: AsPrimitive<i8> + FromPrimitive + Signed,
+// {
+//     fn from(x: I) -> Self {
+//         signed_int_trits(x).collect()
+//     }
+// }
```

Comment:
Is this still relevant ?

Reply:
I'd rather keep for future reference it since it's the 'preferred' way of doing things.

Reply:
Sure